cmd/crossplane/main.go

```diff
@@ -37,48 +36,57 @@ import (
-	)
-	c, ci := core.FromKingpin(app.Command("core", "Start core Crossplane controllers.").Default())
-	r, ri := rbac.FromKingpin(app.Command("rbac", "Start Crossplane RBAC Manager controllers."))
-	cmd := kingpin.MustParse(app.Parse(os.Args[1:]))
+	// // NOTE(negz): We must setup our logger after calling kingpin.MustParse in
+	// // order to ensure the debug flag has been parsed and set.
+	zl := zap.New(zap.UseDevMode(false))
+	ctrl.SetLogger(zl)
 
-	// NOTE(negz): We must setup our logger after calling kingpin.MustParse in
-	// order to ensure the debug flag has been parsed and set.
-	zl := zap.New(zap.UseDevMode(*debug))
-	if *debug {
-		// The controller-runtime runs with a no-op logger by default. It is
-		// *very* verbose even at info level, so we only provide it a real
-		// logger when we're running in debug mode.
-		ctrl.SetLogger(zl)
-	}
 	// Note that the controller managers scheme must be a superset of the
 	// package manager's object scheme; it must contain all object types that
 	// may appear in a Crossplane package. This is because the package manager
 	// uses the controller manager's client (and thus scheme) to create packaged
 	// objects.
 	s := runtime.NewScheme()
-	kingpin.FatalIfError(corev1.AddToScheme(s), "cannot add core v1 Kubernetes API types to scheme")
-	kingpin.FatalIfError(appsv1.AddToScheme(s), "cannot add apps v1 Kubernetes API types to scheme")
-	kingpin.FatalIfError(rbacv1.AddToScheme(s), "cannot add rbac v1 Kubernetes API types to scheme")
-	kingpin.FatalIfError(coordinationv1.AddToScheme(s), "cannot add coordination v1 Kubernetes API types to scheme")
-	kingpin.FatalIfError(extv1.AddToScheme(s), "cannot add apiextensions v1 Kubernetes API types to scheme")
-	kingpin.FatalIfError(extv1beta1.AddToScheme(s), "cannot add apiextensions v1beta1 Kubernetes API types to scheme")
-	kingpin.FatalIfError(apis.AddToScheme(s), "cannot add Crossplane API types to scheme")
 
-	switch cmd {
-	case c.Name:
-		kingpin.FatalIfError(c.Run(s, logging.NewLogrLogger(zl.WithName("core"))), "cannot run crossplane")
-	case ci.Name:
-		kingpin.FatalIfError(ci.Run(s, logging.NewLogrLogger(zl.WithName("core init"))), "cannot initialize crossplane")
-	case r.Name:
-		kingpin.FatalIfError(r.Run(s, logging.NewLogrLogger(zl.WithName("rbac"))), "cannot run RBAC manager")
-	case ri.Name:
-		kingpin.FatalIfError(ri.Run(s, logging.NewLogrLogger(zl.WithName("rbac init"))), "cannot initialize RBAC manager")
-	default:
-		kingpin.FatalUsage("unknown command %s", cmd)
-	}
+	ctx := kong.Parse(&cli,
+		kong.Name("crossplane"),
+		kong.Description("An open source multicloud control plane."),
+		kong.BindTo(logging.NewLogrLogger(zl), (*logging.Logger)(nil)),
+		kong.UsageOnError(),
+		kong.Vars{
+			"rbac_manage_default_var": rbac.ManagementPolicyAll,
+			"rbac_manage_enum_var": strings.Join(
+				[]string{
+					rbac.ManagementPolicyAll,
+					rbac.ManagementPolicyBasic,
+				},
+				", "),
+		},
```

Comment:
Is there any way to attach these vars to the context closer to where they're used, in the RBAC start command?

Reply:
The Vars are options to the parser and attach the vars to the parsed command. I have moved these vars to the RBAC package. Let me know if that would work.